src/main/java/com/contentful/java/cma/ModuleEnvironments.java

```diff
@@ -314,18 +334,40 @@ CMAEnvironment method() {
      * @throws IllegalArgumentException if space id is null.
      * @throws IllegalArgumentException if environment is null.
      */
-    public CMACallback<CMAEnvironment> branch(
+    public CMACallback<CMAEnvironment> clone(
         final CMAEnvironment sourceEnvironment,
         final CMAEnvironment newEnvironment,
         CMACallback<CMAEnvironment> callback) {
       return defer(new DefFunc<CMAEnvironment>() {
         @Override
         CMAEnvironment method() {
-          return ModuleEnvironments.this.branch(sourceEnvironment, newEnvironment);
+          return ModuleEnvironments.this.clone(sourceEnvironment, newEnvironment);
         }
       }, callback);
     }
 
+    /**
+     * Create an environment using a source environment.
+     * <p>
+     * This method will override the configuration specified through
+     * {@link CMAClient.Builder#setSpaceId(String)}.
+     *
+     * @param sourceEnvironment base of the created environment.
+     * @param newEnvironment    to be created environment.
+     * @param callback          Callback
+     * @return the given CMACallback instance
+     * @throws IllegalArgumentException if space id is null.
+     * @throws IllegalArgumentException if environment is null.
+     * @deprecated This method has been renamed to {@link #clone(CMAEnvironment, CMAEnvironment)}.
+     * It will be removed in the next major release.
+     */
+    public CMACallback<CMAEnvironment> branch(
+        final CMAEnvironment sourceEnvironment,
+        final CMAEnvironment newEnvironment,
+        CMACallback<CMAEnvironment> callback) {
+      return this.clone(sourceEnvironment, newEnvironment, callback);
```

Comment:
Please call the non async method here. (Consistency with the calls from above: Those methods deal with asynch and just direct the actual logic further.

```
return ModuleEnvironments.this.branch(sourceEnvironment, newEnvironment);
```

Reply:
Good catch.